wrappers/rust/oneds-telemetry/src/types.rs

```diff
@@ -0,0 +1,182 @@
+use std::{
+    collections::HashMap,
+    ffi::{CStr, CString},
+    os::raw::c_char,
+};
+
+use chrono::{DateTime, Utc};
+use log::{debug, error};
+use once_cell::sync::Lazy;
+
+use cpp_client_telemetry_sys::{evt_prop, evt_prop_v, evt_prop_t_TYPE_STRING, evt_prop_t_TYPE_TIME, evt_prop_t_TYPE_INT64};
+
+#[derive(Clone, Debug)]
+pub struct StringProperty {
+    bytes: Vec<u8>,
+}
+
+impl StringProperty {
+    pub fn new(value: &str) -> Self {
+        let bytes = CString::new(value).expect("Unable to convert value to CString");
+
+        StringProperty {
+            bytes: bytes.to_bytes_with_nul().to_vec(),
+        }
+    }
+
+    pub fn as_ptr(&self) -> *const c_char {
+        self.bytes.as_ptr() as *const c_char
+    }
+
+    pub fn as_bytes(&self) -> Vec<u8> {
+        self.bytes.clone()
+    }
+
+    fn to_str(&self) -> Result<&str, std::str::Utf8Error> {
+        unsafe { CStr::from_ptr(self.as_ptr()).to_str() }
+    }
+}
+
+pub static NAME_PROPERTY: Lazy<StringProperty> = Lazy::new(|| StringProperty::new("name"));
+pub static TIME_PROPERTY: Lazy<StringProperty> = Lazy::new(|| StringProperty::new("time"));
+pub static EMPTY_STRING: Lazy<StringProperty> = Lazy::new(|| StringProperty::new(""));
+
+#[derive(Clone, Debug)]
+pub enum TelemetryProperty {
+    String(StringProperty),
+    Int(i64),
+}
+
+pub type TelemetryPropertyBag = HashMap<&'static str, TelemetryProperty>;
```

Comment:
It's a matter of taste, but I generally prefer to simply see `HashMap<&str, TelemetryProperty>` instead of a type alias.